src/modules/companies/controller.js

```diff
@@ -6,9 +6,26 @@ const companyContactsService = require('../../lib/services/company-contacts');
 
 const documentsHelper = require('./lib/documents');
 const returnsHelper = require('./lib/returns');
-
-const { envelope } = require('../../lib/response');
+const companiesService = require('../../lib/services/companies-service');
+const invoiceAccountsService = require('../../lib/services/invoice-accounts-service');
 const Boom = require('@hapi/boom');
+const { NotFoundError, InvalidEntityError } = require('../../lib/errors');
+const { envelope } = require('../../lib/response');
+
+// caters for error triggered in this service and 404s returned from the CRM
+const isNotFoundError = err => err instanceof NotFoundError || err.statusCode === 404;
+
+const mapErrorResponse = error => {
```

Comment:
Don't know if this `mapErrorResponse` code is similar to that in the billing module - if so we can make a shared one and put in src/lib